cluster/images/etcd/migrate-if-needed.sh

```diff
@@ -46,7 +46,8 @@ set -o nounset
 BUNDLED_VERSIONS="3.0.17, 3.1.12, 3.2.24, 3.3.17, 3.4.7"
 
 # shellcheck disable=SC2039
-ETCD_NAME="${ETCD_NAME:-etcd-${ETCD_HOSTNAME:-$(hostname -s)}}"
+# NOTE: Keep ETCD_NAME in sync with host_name in https://github.com/kubernetes/kubernetes/blob/2e7996e3e2712684bc73f0dec0200d64eec7fe40/cluster/gce/gci/configure-helper.sh#L1563
+ETCD_NAME="${ETCD_NAME:-etcd-${ETCD_HOSTNAME:-$HOSTNAME}}"
```

Comment:
I don't understand why in etcd we set the name to simply "$ETCD_HOSTNAME" and here to "etcd-$ETCD_HOSTNAME".
Shouldn't those two be consistent?

Reply:
etcd name is set to etcd-{{ hostname }} in etcd.manifest, 
https://github.com/kubernetes/kubernetes/blob/b8f24173da300cd8aa6c8daea5d86efcd494b17e/cluster/gce/manifests/etcd.manifest#L27

then hostname is host_name set in configure_helper.sh here: https://github.com/kubernetes/kubernetes/blob/b8f24173da300cd8aa6c8daea5d86efcd494b17e/cluster/gce/gci/configure-helper.sh#L1597

So it is the same here.

Reply:
OK - I get it now.
The problem is that these are not the lines that should be kept in sync. What you really care about is this line:
https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/manifests/etcd.manifest#L27

Reply:
Yes, that makes sense. I removed the added comment in configure-helper.sh and changed this comment here to: "# NOTE: Make sure the resulted ETCD_NAME agrees with --name in etcd.manifest: https://github.com/kubernetes/kubernetes/blob/e7ca64fbe16d0c4b6c7b36aecde9cd75042b2828/cluster/gce/manifests/etcd.manifest#L27". Does this look better now?